data/articles/progress-bar.md

```diff
@@ -0,0 +1,395 @@
+---
+id: 11
+author: "kkpiotrowski"
+topics: ["hanami", "htmx", "sidekiq", "rom-rb", "redis", "sequel"]
+title: "Hanami and HTMX progress bar"
+excerpt: "<<Summary>>"
+videoId: null
+published: false
+publishedAt: 
+modifiedAt: 
+aliases: ['HMEP011']
+thumbnail:
+  full: https://via.placeholder.com/1920/600?text=HanamiMastery.com
+  big: https://via.placeholder.com/780/440?text=HanamiMastery.com
+  small: https://via.placeholder.com/360/200?text=HanamiMastery.com
+discussions:
+  twitter: null
+  reddit:
+    ruby: null
+    rails: null
+    hanamirb: null
+source: 
```

Comment:
Can you give me a pointer to the source code? `Before vs After` diff would be amazing.

Reply:
https://github.com/krzykamil/libus/commits/master/ This is the full repo
I haven't done it all in one commit, since it would be huge (and I was trying different approaches along the way), but the most important commit is this one, since it is the gone that gives a full working state for the first time, with the persistence layer added https://github.com/krzykamil/libus/commit/4abab20af5f5d6faae20bcaf9c8ca4975e7aaba9
The commits after that are refactors, cleanup and rom-factory setup